client/components/credentials-form/index.jsx

```diff
@@ -0,0 +1,139 @@
+import LoggedOutForm from 'components/logged-out-form';
+import LoggedOutFormFooter from 'components/logged-out-form/footer';
+import { getConnectingSite } from 'state/jetpack-connect/selectors';
+
+class CredentialsForm extends Component {
+	static propTypes = {
+		submitButtonText: PropTypes.string.isRequired,
+	};
+
+	state = {
+		notice: null,
+		submitting: false,
+		form: null,
+		signedUp: false,
+		validationInitialized: false,
+	};
+
+	getInitialFields() {
+		return {
+			username: '',
+			password: '',
+			submitting: false,
+		};
+	}
+
+	sanitizeUsername( username ) {
+		return username && username.replace( /[^a-zA-Z0-9]/g, '' ).toLowerCase();
+	}
+
+	sanitize = ( fields, onComplete ) => {
+		const sanitizedUsername = this.sanitizeUsername( fields.username );
+
+		if ( fields.username !== sanitizedUsername ) {
+			onComplete( {
+				username: sanitizedUsername,
+			} );
+		}
+	};
+
+	handleSubmit = event => {
+		event.preventDefault();
+
+		if ( this.state.submitting ) {
+			return;
+		}
+		this.setState( { submitting: true } );
+
+		if ( this.props.installJetpack ) {
+			const url = this.props.jetpackConnectSite.url;
+			const user = this.state.user;
+			const password = this.state.password;
+			this.props.actionOnSubmit( url, user, password );
+
+			return;
+		}
+		// to make the form flexible, we allow to pass extra actions
+		this.props.actionOnSubmit();
+	};
+
+	getChangeHandler = field => event => {
+		this.setState( { [ field ]: event.target.value } );
+	};
+
+	formFields() {
+		return (
+			<div>
+				<FormLabel htmlFor="username">{ this.props.translate( 'Username' ) }</FormLabel>
+				<FormTextInput
+					autoCapitalize="off"
+					autoCorrect="off"
+					className="credentials--form__input"
+					disabled={ this.state.submitting }
+					id="username"
+					name="username"
+					onChange={ this.getChangeHandler( 'username' ) }
+					value={ this.state.username || '' }
+				/>
+
+				<FormLabel htmlFor="password">{ this.props.translate( 'Password' ) }</FormLabel>
+				<FormPasswordInput
+					className="credentials--form__input"
+					disabled={ this.state.submitting }
+					id="password"
+					name="password"
+					onChange={ this.getChangeHandler( 'password' ) }
+					value={ this.state.password || '' }
+				/>
+			</div>
+		);
+	}
+
+	formFooter() {
+		return (
+			<LoggedOutFormFooter>
+				<FormButton
+					className="credentials--form__submit"
+					disabled={ ! this.state.username || ! this.state.password }
+				>
+					{ this.props.submitButtonText }
+				</FormButton>
+			</LoggedOutFormFooter>
+		);
+	}
+
+	render() {
+		return (
+			<div className="credentials-form">
+				<LoggedOutForm onSubmit={ this.handleSubmit }>
+					{ this.formFields() }
+					{ this.props.formFooter || this.formFooter() }
+				</LoggedOutForm>
+				{ this.props.footerLink }
+			</div>
+		);
+	}
+}
+
+export default connect( state => {
+	return { jetpackConnectSite: getConnectingSite( state ) };
```

Comment:
If this is going to be a generic component (in /client/components), we should keep all the jetpack-specific stuff (url, action) in the parent component, and pass an `onSubmit()` function as a prop to this component.

Reply:
Actually, I might be tempted to do everything in one, custom component. I'm not sure of the value of having the entry form part split out. Nothing wrong with the way it is though, and maybe it could be reused—I don't know.